languages/ruby/lib/oso/oso.rb

```diff
@@ -19,5 +34,143 @@ def initialize
     def allowed?(actor:, action:, resource:)
       !query_rule('allow', actor, action, resource).first.nil?
     end
+
+    # Ensure that +actor+ is allowed to perform +action+ on
+    # +resource+.
+    #
+    # If the action is permitted with an +allow+ rule in the policy, then
+    # this method returns +None+. If the action is not permitted by the
+    # policy, this method will raise an error.
+    #
+    # The error raised by this method depends on whether the actor can perform
+    # the +"read"+ action on the resource. If they cannot read the resource,
+    # then a {Oso::NotFoundError} error is raised. Otherwise, a
+    # {Oso::ForbiddenError} is raised.
+    #
+    # @param actor The actor performing the request.
+    # @param action The action the actor is attempting to perform.
+    # @param resource The resource being accessed.
+    # @param check_read [Boolean] If set to +false+, a {Oso::ForbiddenError} is
+    #   always thrown on authorization failures, regardless of whether the actor
+    #   can read the resource. Default is +true+.
+    #
+    # @raise [Oso::ForbiddenError] Raised if the actor does not have permission
+    #   to perform this action on this resource, but _does_ have +"read"+
+    #   permission on the resource.
+    # @raise [Oso::NotFoundError] Raised if the actor does not have permission
+    #   to perform this action on this resource and additionally does not have
+    #   permission to +"read"+ the resource.
+    def authorize(actor, action, resource, check_read: true)
+      return if query_rule_once('allow', actor, action, resource)
```

Comment:
I don't know about this. I suppose there is an argument to be made, but I'd rather not impose a dependency on that particular public API.

Also note that its actually more verbose because you have to do `allowed?(actor: actor, action: action, resource: resource)`